### ISSUE TITLE
Listen on port 443

### ISSUE THREAD

**Issue by tinyhammers** [netdata/netdata#7219]
Hello hoomans...

I'm back with another n00b question. Brace yourselves...

I'm trying to get all our slaves to stream to our master over SSL. 
I've read https://github.com/netdata/netdata/tree/master/streaming and https://github.com/netdata/netdata/tree/master/web/server#enabling-tls-support, but I'm still stumped.
I'm testing it on one slave, and have changed the stream destination to XXX.XXX.XXX.XXX:443 and the error logs are giving me this little nugget:
`2019-10-29 11:47:29: netdata ERROR : STREAM_SENDER[mah-hostname] : STREAM mah-hostname [send to XXX.XXX.XXX.XXX:443]: failed to connect`

That suggests that my slave is attempting to connect correctly to my master on the correct port...HOWEVER... I can't get the master to listen on port 443.

I've added the paths to the cert and key in the `[web]` section of the config (these have been setup by a colleague who knows what the hell he is doing, unlike myself), as per the instructions, and my understanding of said instructions is that that is all I need to do.

If I run `netstat -plntd` I get this:
`tcp        0      0 0.0.0.0:19999           0.0.0.0:*               LISTEN      25208/netdata `

Y U NO LISTEN ON 443?

Is it something to do with the `bind to` setting? Every time I change anything there, Netdata fails to start.
I guess I'm clearly misunderstanding something crucial here and I throw myself upon your merciful selves for some assistance.

Thanks, you 🤗

**Comment by tinyhammers** [comment]
Oh, just to add that streaming works fine via http.

**Comment by thiagoftsm** [comment]
Hi @tinyhammers,

For I help you correctly, let me do some questions for you:

1. Did you set key and certificate on master
```
        ssl key = /etc/netdata/ssl/key2048.pem
        ssl certificate = /etc/netdata/ssl/cert2048.pem
```
?
2. Netdata will always listen the port 19999, unless you change the parameter `default port`. Did you change it? Please, pay attention for the fact that to set SSL does not mean begin to use 443, because we can add TLS for any port.
3. In the client side, it will be necessary to a destination variable like this:
```
destination = 192.168.0.1:19999:SSL
```
Here I am considering that your Netdata master is using the IP 192.168.0.1.

I hope with this you can move in front, but case you have more doubts, please, let us know.

Best regards!

**Comment by ilyam8** [comment]
> I can't get the master to listen on port 443.
> SSL does not mean begin to use 443

Yes, port it is just a number. Another reason you dont need to change the port to 443 is that only root user allowed to listen to ports below 1024.

**Comment by tinyhammers** [comment]
Oh @thiagoftsm, the cert files are in a different directory `/etc/letsencrypt/live/netdata-master.my-domain.com`
Do they necessarily need to be in `/etc/netdata/ssl`, or can they be somewhere else as long the config file has the correct path?

**Comment by thiagoftsm** [comment]
> Oh @thiagoftsm, the cert files are in a different directory `/etc/letsencrypt/live/netdata-master.my-domain.com`
> Do they necessarily need to be in `/etc/netdata/ssl`, or can they be somewhere else as long the config file has the correct path?

No, you can put store the files in the better directory for you, but Netdata must have permission to read it, please, check in your error.log, in the master netdata, whether you have the following message:

```
SSL cannot check the private key: ...
```

**Comment by thiagoftsm** [comment]
It is possible that the description I gave in the previous message has a direct relationship with the error that you reported.

**Comment by tinyhammers** [comment]
UPDATE: FIXED IT!

So tried running Netdata behind nginx, but it transpires that this only works for the dashboard, not the streaming, so that's good to know :)

Turns out the problem was due to my boss having NOT SET THE SSL CERTS UP PROPERLY.

So on the plus side, I may be able to take his job as I may very well "[send him to Belize](https://www.urbandictionary.com/define.php?term=send%20him%20to%20belize)", for causing me days of headaches.

As always, thank you, thank you, thank you, for giving me free help for this free thing you've made ❤️.
